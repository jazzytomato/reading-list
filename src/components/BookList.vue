<template>
  <div class="avenir dark-gray pa4 f3 f4-ns fw4">
    <h1>Books</h1>
    <div v-if="store.state.books.isFetching">
      Loading...
    </div>
    <div v-if="store.state.books.isError">
      Error while fetching book data.
    </div>
    <div v-if="store.state.books.isLoaded">
      <ul class="pl0 list flex flex-column items-start">
        <BookListItem v-for="book in store.state.books.data" v-bind="book" v-bind:key="book.id"/>
      </ul>
    </div>
  </div>
</template>

<script>
import { store } from '@/store.js'
import BookListItem from '@/components/BookListItem'

export default {
  name: 'BookList',
  data () {
    return { store }
  },
  mounted: () => {
    store.fetchBooksIfNeeded()
  },
  components: {BookListItem}
}
</script>
