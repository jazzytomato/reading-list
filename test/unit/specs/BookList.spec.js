// import Vue from 'vue'
// import BookList from '@/components/BookList'
// jest.mock('fetch', () => ({
//     get: jest.fn(() => Promise.resolve({ data: 3 }))
//   }));

// import fetch from 'fetch'

// describe('BookList.vue', () => {
//   it('should render correct contents', () => {
//     const Constructor = Vue.extend(BookList)
//     const vm = new Constructor().$mount()
//     expect(vm.$el.querySelector('h1').textContent)
//       .toEqual('Books')
//   })
// })

import { shallowMount } from '@vue/test-utils'
import fetchMock from 'fetch-mock'
import flushPromises from 'flush-promises'
import BookList from '@/components/BookList.vue'

const apiData = {
  books: []
}

describe('BookList.vue', () => {
  beforeEach(() => {
    fetchMock.get('http://www.mocky.io/v2/5c30dff23000006800e77ace', apiData)
  })

  it('renders empty component', async () => {
    const wrapper = shallowMount(BookList)
    await flushPromises()
    expect(wrapper.element).toMatchSnapshot()
  })

  afterEach(() => {
    fetchMock.restore()
  })
})
