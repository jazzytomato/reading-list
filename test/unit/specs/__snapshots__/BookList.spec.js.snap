// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList.vue renders empty component 1`] = `
<div
  class="avenir dark-gray pa4 f3 f4-ns fw4"
>
  <h1>
    Books
  </h1>
   
  <!---->
   
  <!---->
   
  <div>
    <ul
      class="pl0 list flex flex-column items-start"
    />
  </div>
</div>
`;
